spec/controllers/api/v1/notes_controller_spec.rb

```diff
@@ -0,0 +1,132 @@
+    let(:user_notes) { create_list(:note, user_note_count, user: user) }
+    let(:expected_keys) { %w[id title note_type content_length] }
+
+    context 'when user is logged in' do
+      include_context 'with authenticated user'
+
+      before do
+        user_notes
+        get :index, params: params
+      end
+
+      context 'when fetching all user notes' do
+        let(:params) { nil }
+
+        include_examples 'successful response'
+      end
+
+      context 'when fetching notes with page paramaters' do
+        let(:page) { 1 }
+        let(:page_size) { Faker::Number.between(from: 1, to: user_note_count) }
+        let(:params) { { page: page, page_size: page_size } }
+
+        include_examples 'successful response'
+
+        it 'retrieves the correct number of notes' do
+          expect(response_body.size).to eq(page_size)
+        end
+      end
+
+      context 'when fetching notes using filters' do
+        let(:filtered_notes) { user_notes.select { |note| note.note_type == note_type } }
+        let(:params) { { note_type: note_type } }
+
+        context 'when note_type is valid' do
+          let(:note_type) { %w[review critique].sample }
+
+          include_examples 'successful response'
+
+          it 'retrieves only the notes of the note_type' do
+            expect(response_body.size).to eq(filtered_notes.size)
+          end
+        end
+
+        context 'when note_type is invalid' do
+          let(:note_type) { 'invalid_note_type' }
+
+          it 'returns an unprocessable entity status' do
+            expect(response).to have_http_status(:unprocessable_entity)
+          end
+        end
+      end
+
+      context 'when fetching notes with order' do
+        context 'when order is asc' do
+          let(:notes_sorted_asc) { user_notes.sort_by(&:created_at) }
+          let(:params) { { order: 'asc' } }
+
+          it 'returns notes ordered upward' do
+            expect(response_body.first['id']).to eq(notes_sorted_asc.first['id'])
+          end
+        end
+
+        context 'when order is desc' do
+          let(:notes_sorted_desc) { user_notes.sort_by(&:created_at).reverse }
+          let(:params) { { order: 'desc' } }
+
+          it 'returns notes ordered descendant' do
+            expect(response_body.first['id']).to eq(notes_sorted_desc.first['id'])
+          end
+        end
+      end
+    end
+
+    context 'when user is not logged in' do
+      before { get :index }
+
+      it_behaves_like 'unauthorized'
+    end
+  end
+
+  describe 'GET #show' do
+    context 'when user is logged in' do
+      include_context 'with authenticated user'
+      let(:expected) { ShowNoteSerializer.new(note).to_json }
+
+      before { get :show, params: params }
+
+      context 'when fetching a valid note' do
+        let(:note) { create(:note, user: user) }
+        let(:params) { { id: note.id } }
+
+        it 'responds with the correct number of items' do
```

Comment:
Que seria el "correct number of items", los atributos? Ojo que el show siempre devuelve una unica nota. 
El texto de este it puede generar dudas de que hace el index

Reply:
Cambie el titulo del it ya que no reflejaba correctamente el funcionamiento de este test, ademas agregue un test para que veriffique que la nota que trae show sea la misma del id